sycl/test/extensions/device_global/device_global_properties.cpp

```diff
@@ -0,0 +1,106 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsyntax-only -Xclang -verify -Xclang -verify-ignore-unexpected=note,warning %s
+// expected-no-diagnostics
+
+#include <CL/sycl.hpp>
+
+using namespace sycl::ext::oneapi::experimental;
+
+static device_global<int> DeviceGlobal1;
+static device_global<int, decltype(properties(device_image_scope))>
+    DeviceGlobal2;
+static device_global<int, decltype(properties(host_access_none))> DeviceGlobal3;
+static device_global<int, decltype(properties(init_mode_reset))> DeviceGlobal4;
+static device_global<int, decltype(properties(implement_in_csr_on))>
+    DeviceGlobal5;
+static device_global<int, decltype(properties(
+                              implement_in_csr_off, host_access_write,
+                              device_image_scope, init_mode_reprogram))>
+    DeviceGlobal6;
+
+// Checks is_property_key_of and is_property_value_of for T.
+template <typename T> void checkIsPropertyOf() {
+  static_assert(is_property_key_of<device_image_scope_key, T>::value);
+  static_assert(is_property_key_of<host_access_key, T>::value);
+  static_assert(is_property_key_of<init_mode_key, T>::value);
+  static_assert(is_property_key_of<implement_in_csr_key, T>::value);
+
+  static_assert(is_property_value_of<decltype(device_image_scope), T>::value);
+  static_assert(is_property_value_of<decltype(host_access_read), T>::value);
+  static_assert(is_property_value_of<decltype(host_access_write), T>::value);
+  static_assert(
+      is_property_value_of<decltype(host_access_read_write), T>::value);
+  static_assert(is_property_value_of<decltype(host_access_none), T>::value);
+  static_assert(is_property_value_of<decltype(init_mode_reset), T>::value);
+  static_assert(is_property_value_of<decltype(init_mode_reprogram), T>::value);
+  static_assert(is_property_value_of<decltype(implement_in_csr_on), T>::value);
+  static_assert(is_property_value_of<decltype(implement_in_csr_off), T>::value);
+}
+
+int main() {
+  static_assert(is_property_key<device_image_scope_key>::value);
+  static_assert(is_property_key<host_access_key>::value);
+  static_assert(is_property_key<init_mode_key>::value);
+  static_assert(is_property_key<implement_in_csr_key>::value);
+
+  static_assert(is_property_value<decltype(device_image_scope)>::value);
```

Comment:
Would something like this be applicable yet readable enough:

```
auto AssertIsValue() = [](auto Prop) {
  static_assert(is_property_value<decltype(Prop)>::value);
}
```

?

Reply:
I don't mind it, but I think the current version is better for also serving as an example for how to use `is_property_value`.